processor/elasticinframetricsprocessor/config_test.go

```diff
@@ -0,0 +1,38 @@
+package elasticinframetricsprocessor
+
+import (
+	"testing"
+)
+
+// TestConfigValidate tests the Validate method of the Config struct.
+func TestConfigValidate(t *testing.T) {
+	tests := []struct {
+		name    string
+		config  Config
+		wantErr bool
+	}{
+		{
+			name:    "Default configuration",
+			config:  Config{},
+			wantErr: false,
+		},
+		{
+			name:    "AddSystemMetrics true",
+			config:  Config{AddSystemMetrics: true},
+			wantErr: false,
+		},
+		{
+			name:    "AddSystemMetrics false",
+			config:  Config{AddSystemMetrics: false},
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := tt.config.Validate(); (err != nil) != tt.wantErr {
```

Comment:
What do you think of testing the configuration by initializing the factory instead? That way, we will be able to use upstream helper functions to validate the configuration `component.ValidateConfig`.

Example: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/k8sattributesprocessor/config_test.go#L194

In addition, I would suggest using the `testdata` directory to include some YAML configurations which will test the unmarshalling phase too.

Let me know if you want me to provide more details/help with the implementation.

Reply:
I have used the upstream validators and the testdata folder now. 
Please see if this looks fine.